opm/input/eclipse/Schedule/ScheduleGrid.hpp

```diff
@@ -36,21 +37,42 @@ namespace Opm {
 class ScheduleGrid
 {
 public:
+    ScheduleGrid(const EclipseGrid& ecl_grid,
+                 const FieldPropsManager& fpm,
+                 CompletedCells& completed_cells,
+                 std::vector<CompletedCells>& completed_cells_lgr,
+                 const std::unordered_map<std::string, std::size_t>& label_to_index_);
+    ScheduleGrid(CompletedCells& completed_cells, 
+                 std::vector<CompletedCells>& completed_cells_lgr,
+                 const std::unordered_map<std::string, std::size_t>& label_to_index_);
     ScheduleGrid(const EclipseGrid& ecl_grid,
                  const FieldPropsManager& fpm,
                  CompletedCells& completed_cells);
+    explicit ScheduleGrid(Opm::CompletedCells& completed_cells);
 
-    explicit ScheduleGrid(CompletedCells& completed_cells);
-
+    ~ScheduleGrid() = default;
     const CompletedCells::Cell&
     get_cell(std::size_t i, std::size_t j, std::size_t k) const;
-
+    const CompletedCells::Cell&
+    get_cell(std::size_t i, std::size_t j, std::size_t k, std::optional<std::string> tag) const;
+    const CompletedCells::Cell&
+    get_cell_lgr(std::size_t i, std::size_t j, std::size_t k, std::string tag) const; 
     const Opm::EclipseGrid* get_grid() const;
+    int get_lgr_grid_number(std::optional<std::string> lgr_label) const;
 
 private:
+    bool lgr_intialized{false};
     const EclipseGrid* grid{nullptr};
     const FieldPropsManager* fp{nullptr};
-    CompletedCells& cells;
+    Opm::CompletedCells& cells; 
```

Comment:
We're already in the `Opm` namespace so there's no need to use the `Opm::` namespace prefix here.